pyvista/__init__.py

```diff
@@ -8,27 +8,323 @@
+    VtkErrorCatcher,
+    Wavelet,
+    XMLImageDataReader,
+    XMLMultiBlockDataReader,
+    XMLPImageDataReader,
+    XMLPRectilinearGridReader,
+    XMLPUnstructuredGridReader,
+    XMLPolyDataReader,
+    XMLRectilinearGridReader,
+    XMLStructuredGridReader,
+    XMLUnstructuredGridReader,
+    # needs to be available per contract with Sphinx-Gallery:
+    _get_sg_image_scraper,
     _get_vtk_id_type,
-    vtk_version_info,
     _set_plot_theme_from_env,
+    abstract_class,
+    array_from_vtkmatrix,
+    # arrays,
+    assert_empty_kwargs,
+    axis_rotation,
+    cell_array,
+    cells,
+    check_depth_peeling,
+    check_valid_vector,
+    # common,
+    compare_images,
+    conditional_decorator,
+    convert_array,
+    convert_string_array,
+    create_grid,
+    cubemap,
+    cubemap_from_filenames,
+    # errors,
+    # features,
+    field_array,
+    fileio,
+    fit_plane_to_points,
+    from_meshio,
+    generate_plane,
+    # geometric_objects,
+    get_array,
+    get_array_association,
+    get_ext,
+    get_gpu_info,
+    get_reader,
+    get_vtk_type,
+    grid_from_sph_coords,
+    helpers,
+    image_to_texture,
+    is_inside_bounds,
+    is_meshio_mesh,
+    is_pyvista_dataset,
+    line_segments_from_points,
+    lines_from_points,
+    make_tri_mesh,
+    merge,
+    # misc,
+    numpy_to_texture,
+    parametric_keywords,
+    # parametric_objects,
+    parse_field_choice,
+    perlin_noise,
+    point_array,
+    raise_not_matching,
+    read,
+    read_exodus,
+    read_legacy,
+    read_meshio,
+    read_plot3d,
+    read_texture,
+    reader,
+    # regression,
+    row_array,
+    sample_function,
+    save_meshio,
+    send_errors_to_logging,
+    set_default_active_scalars,
+    set_default_active_vectors,
+    set_error_output_file,
     set_pickle_format,
+    set_vtkwriter_mode,
+    # sphinx_gallery,
+    start_xvfb,
+    surface_from_para,
+    threaded,
+    trans_from_matrix,
+    transform_vectors_sph_to_cart,
+    transformations,
+    translate,
+    try_callback,
+    vector_poly_data,
+    voxelize,
+    vtk_bit_array_to_char,
+    vtk_id_list_to_array,
+    vtk_points,
+    vtk_version_info,
+    vtkmatrix_from_array,
+    wrap,
+    # wrappers,
+    # xvfb,
+)
+
+from pyvista.core import (
+    CellType,
+    CompositeFilters,
+    DataObject,
+    DataSet,
+    DataSetAttributes,
+    DataSetFilters,
+    ExplicitStructuredGrid,
+    Grid,
+    MultiBlock,
+    PointGrid,
+    PointSet,
+    PolyData,
+    PolyDataFilters,
+    RectilinearGrid,
+    StructuredGrid,
+    Table,
+    Texture,
+    UniformGrid,
+    UniformGridFilters,
+    UnstructuredGrid,
+    UnstructuredGridFilters,
+    # cell,
+    # composite,
+    # dataobject,
+    # dataset,
+    # datasetattributes,
+    errors,
+    filters,
+    # grid,
+    # objects,
+    # pointset,
 )
+
+# mypy gets confused if this is added to the pyvista.core import:
+from pyvista.core.pyvista_ndarray import pyvista_ndarray
```

Comment:
This is a major thorn in my side, but if I add `pyvista_ndarray` to the preceding block of imports (where it belongs) `mypy` thinks that what gets imported is the `pyvista.core.pyvista_ndarray` _module_ rather than the `pyvista.core.pyvista_ndarray.pyvista_ndarray` _class_. Despite the fact that in `pyvista/core/__init__.py` it correctly finds that `from .pyvista_ndarray import pyvista_ndarray` refers to the class, not the module. So this seems like a `mypy` issue. I have no other idea to work around this than to use this ugly fully qualified import in `pyvista/__init__.py`.

Reply:
I doubt anyone in the wild has done `from pyvista.core.pyvista_ndarray import pyvista_ndarray` (right?) in which case you could `git mv pyvista/core/pyvista_ndarray.py pyvista/core/_pyvista_ndarray.py` and then adjust the imports. Seems cleaner anyway since `pyvista.core.pyvista_ndarray` is not meant to be a public module I assume...

Reply:
Yeah, we're currently offering a very broad API surface (mostly unintentionally I think). These are partly the kinds of "public API paring down" steps I have in mind, but need to discuss in detail to make sure we're all on board and don't break downstream (or at least are ready to do that break and urge downstream to use proper import paths).